packages/waku/src/router/base-types.ts

```diff
@@ -0,0 +1,22 @@
+import type { createPages } from './create-pages.js';
+import type { CollectPaths } from './inferred-router-types.js';
+
+type AnyRouter = ReturnType<typeof createPages>;
+
+type RouterSpec<Router extends AnyRouter> = Awaited<
+  NonNullable<Router['DO_NOT_USE_ready']>
+>[number];
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface RouteConfig {
+  // routes to be overridden by users
+}
+
+export type PathsForRouter<Router extends AnyRouter> =
+  CollectPaths<RouterSpec<Router>> extends never
+    ? string
+    : CollectPaths<RouterSpec<Router>> & {};
```

Comment:
This should probably move to `create-pages.ts`

Reply:
this generic needs to be exported from one of our modules for users to be able to do

```ts
paths: PathsForRouter<typeof myRouter>
```

I could move this to `createPages`, but then would need to re-export through another file?

Reply:
yes, it should be exported through `create-pages.ts`.
I was thinking exporting `waku/router/server`, but hmm, yeah `waku/router` should be fine.
Just re-export directly like `export type { ... } from './create-pages.js'`;.

Let me re-consider about `base-types.ts` naming afterwards.

Reply:
Moved this to `create-pages/inferred-router-types.ts` and re-exported from `base-types.ts` 👍